src/styles/main.scss

```
@@ -1,3 +1,62 @@
 body {
-  margin: 0;
+  margin: 50px;
+}
+
+$cell-size: 100px;
+$gap: 1px;
+$day-name: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($cell-size+$gap)*7;
+  box-sizing: border-box;
+  gap: $gap;
+
+  &__day {
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    border: 1px solid;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    cursor: pointer;
+    transition: background-color, transform 0.5s;
+  }
+
+  &__day:hover {
+    background-color: rgb(250, 5, 250);
+    transform: translateY(-20px);
+  }
+
+  &::before {
+    display: block;
+  }
+
+  @for $day from 1 through 31 {
+    &__day:nth-child(#{$day})::before {
+      content: "#{$day}";
+    }
+  }
+
+  @each $day, $index in $day-name {
+    &--start-day-#{$day} :first-child {
+      margin-left: $index * ($cell-size + $gap);
+    }
+  }
+
+  @for $month-length from 28 through 31 {
+    &--month-length--#{$month-length} > :not(:nth-child(-n+#{$month-length})) {
+      visibility: hidden;
```

Consider it will be better
```suggestion
      display: none;
```

---

done